raiden/transfer/channel.py

```diff
@@ -2747,12 +2770,12 @@ def handle_channel_batch_unlock(
             partner_state.onchain_locksroot = Locksroot(LOCKSROOT_OF_NO_LOCKS)
 
         # only clear the channel state once all unlocks have been done
-        no_unlock_left_to_do = our_state.onchain_locksroot == Locksroot(
-            LOCKSROOT_OF_NO_LOCKS
-        ) and partner_state.onchain_locksroot == Locksroot(LOCKSROOT_OF_NO_LOCKS)
-
-        if no_unlock_left_to_do:
-            new_channel_state = None
+        # no_unlock_left_to_do = our_state.onchain_locksroot == Locksroot(
```

Comment:
https://github.com/raiden-network/raiden/pull/6400/files#diff-fe7772da6084a3493b31e88fbf7f6d55R2761 probably needs to be changed to allow unlock in a state other than `SETTLED`